client/src/hooks/Result.tsx

```diff
@@ -0,0 +1,49 @@
+import { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ClassNameMap } from '@material-ui/styles/withStyles';
+import { detectItemType } from '@/utils/Common';
+import CopyButton from '@/components/advancedSearch/CopyButton';
+
+export const useSearchResult = (searchResult: any[], classes: ClassNameMap) => {
+  const { t: commonTrans } = useTranslation();
+  const copyTrans = commonTrans('copy');
+
+  return useMemo(
+    () =>
+      searchResult?.map((resultItem: { [key: string]: any }) => {
+        // Iterate resultItem keys, then format vector(array) items.
+
+        const tmp = Object.keys(resultItem).reduce(
+          (prev: { [key: string]: any }, item: string) => {
+            const itemType = detectItemType(resultItem[item]);
+
+            switch (itemType) {
+              case 'json':
```

Comment:
Should we use enum for these types?

Reply:
I don't think so, string is enough here.